src/tdastro/astro_utils/snia_utils.py

```diff
@@ -0,0 +1,221 @@
+import numpy as np
+from astropy.cosmology import FlatLambdaCDM
+from scipy.stats import norm
+from scipy.stats.sampling import NumericalInversePolynomial
+
+from tdastro.base_models import FunctionNode
+
+
+class HostmassX1Distr:
+    """
+    A class that contains the pdf of the SALT x1 parameter given the hostmass
+
+    Attributes
+    ----------
+    hostmass: `float`
+        The hostmass value.
+
+    Parameters
+    ----------
+    hostmass: `float`
+        The hostmass value.
+    """
+
+    def __init__(self, hostmass):
+        self.hostmass = hostmass
+
+    def _p(self, x1, hostmass=9.0):
+        """
+        The probablity of having a value of x1 given a hostmass.
+
+        Parameters
+        ----------
+        x1: `float`
+            The x1 value.
+        hostmass: `float`
+            The hostmass value.
+
+        Returns
+        -------
+        p: `float`
+            The probablity.
+        """
+
+        p = 0.0 if x1 < -5 or x1 > 5 else (np.exp(-(x1**2)) if x1 < 0.0 else 1.0) if hostmass < 10.0 else 1.0
```

Comment:
Should we vectorize it?
```suggestion
        p = np.exp(-(np.max(0, x1)**2)
        p = np.where(np.logical_and(x1 >= -5, x1 <= 5), p, 0.0)
        p = np.where(hostmass < 10.0, p, 1.0)
```